File: chat-uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/OnModifyMessageListener.java
package io.agora.chat.uikit.chat.interfaces;


import io.agora.chat.ChatMessage;

/**
 * The user listens for modifing messages successfully
 */
public interface OnModifyMessageListener {
    /**
     * modify message success
     * @param messageModified
     */
    void onModifyMessageSuccess(ChatMessage messageModified);

    /**
     * modify message failure
     * @param messageId
     * @param code
     * @param error
     */
    void onModifyMessageFailure(String messageId, int code, String error);
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/interfaces/OnModifyMessageListener.kt
package io.agora.uikit.feature.chat.interfaces

import io.agora.uikit.common.ChatMessage

/**
 * The user listens for modifing messages successfully
 */
interface OnModifyMessageListener {
    /**
     * modify message success
     * @param messageModified
     */
    fun onModifyMessageSuccess(messageModified: ChatMessage?)

    /**
     * modify message failure
     * @param messageId
     * @param code
     * @param error
     */
    fun onModifyMessageFailure(messageId: String?, code: Int, error: String?)
}